### ISSUE TITLE
New M1 Pro MacBook Pro - Keeps asking to allow Karabiner-VirtualHIDDevice-Manager, even after allowing

### ISSUE THREAD

**Issue by EnygmaINC** [pqrs-org/Karabiner-Elements#3368]
macOS version: macOS 13.1
Karabiner version: Karabiner-Elements 14.11.0
Your Mac hardware: MacBook Pro (2021) - M1 Pro
Your keyboard hardware: HyperX Alloy Origins Core

Just setting this up now, coming from a 2016 MacBook Pro w/ Intel Core i7 - allowed system extensions via rebooting and allowing lower security, etc. etc. Hit "allow" when prompted to. Kept login items as recommended to do. Tried uninstalling/reinstalling multiple times, nothing. Tried safe mode, "Allow" didn't show again after the initial time, yet will not work- keeps prompting for the same **Karabiner-VirtualHIDDevice-Manager**- completely out of my wits here... what am I doing wrong?

**Comment by lrybak** [comment]
I have same problem. I have started troubleshooting issue with not working Complex Modifications. I noticed that each time Karabiner starts it displays warning about not loaded Karabiner-VirtualHIDDevice-Manager. The alert message litarally blink for maybe second so I'm not able to follow [instruction](https://karabiner-elements.pqrs.org/docs/help/troubleshooting/driver-alert-keeps-showing-up/) to deactivate Driver.
I followed steps mentioned by @EnygmaINC but nothing happens. After reinstallation and reboot Complex Modifications do not work and Karabiner still shows Karabiner-VirtualHIDDevice-Manager loading warning.
Any ideas how to fix that?

**Comment by olliepop777** [comment]
Does this prevent Karabiner Elements from functioning correctly for you? Every time I launch the app, I get the screen asking my to allow it but then after a second it goes away. And then the app seems to function normally. I'm not sure if you are talking about something else though

**Comment by Seigiard** [comment]
Faced same problem. Didn't find a solution

**Comment by piskov** [comment]
Yep, definitely something like that also happened to me.

One of the “quirks” was that after Complex Modifications stopped working, I coudn’t find Karabiner Elements via Spotlight (only Event Viewer showed up). Turned out Karabiner was replaced as a Alias (shortcut in Windows terminlogy) in the Applications folder.

I don't remember specific steps I took to remedy the situation but some of the included:
- downloading Karabiner installer manually from the website;
- reinstalling (I don't remember whether I uninstalled it via Karabiner UI “Uninstall” button);
- removing (with minus UI-button) 3 karabiner items in the macOS Input Monitoring privacy section (although I don't remebmer whether that was before reinstall or after);
- plus, couple of reboots.

TLDR; it took several minutes of mostly autopilot actions but it works.

Just for the sake of full disclosure: I have latest macOS public beta (13.2) and always use latest Karabiner beta (“Check for beta updates” button in the Karabiner settings).

**Comment by EnygmaINC** [comment]
@piskov 

Tried all of these myself - alas, the beta = release version, so no joy there. 

@tekezo - any idea?

**Comment by Seigiard** [comment]
@EnygmaINC  same story.

- Notice that `systemextensionsctl` shows Karabiner's extension
- Disable SIP protection via Recovery Mode and `csrutil disable`
- Remove this extention: `systemextensionsctl uninstall G43BCU2T37  org.pqrs.Karabiner-DriverKit-VirtualHIDDevice` & `systemextensionsctl reset`
- Enable SIP protection via Recovery Mode and `csrutil enable`

- Completely [remove Karabiner](https://karabiner-elements.pqrs.org/docs/manual/operation/uninstall/#additional-uninstallation) with all log folders

- Check that `systemextensionsctl list` doesn't show any extension
- Install Karabiner via `brew install --cask karabiner-elements`
- Karabiner still displays warning about not loaded Karabiner-VirtualHIDDevice-Manager
- Karabiner doesn't work
- Remove Karabiner

- Check that `systemextensionsctl list` doesn't show any extension
- Install Karabiner from site
- Karabiner still displays warning about not loaded Karabiner-VirtualHIDDevice-Manager
- Karabiner doesn't work
- Remove Karabiner

**Comment by Seigiard** [comment]
@EnygmaINC @lrybak 

In my case problem was in a symlink **from Dropbox folder.** 
Check this comment for more info: https://github.com/pqrs-org/Karabiner-Elements/issues/3338#issuecomment-1357781203

Hope it helps you too

**Comment by gzz2000** [comment]
Same problem here. Waiting for a solution.
I upgraded from macOS 11.2.3 to 13.2. Before the upgrade it worked fine. However, it keeps asking me to allow the system extension even if I have allowed that many times. Rebooting, reinstalling, and safe mode don't work.

**Comment by gzz2000** [comment]
Looking into the console, one message that might be of interest:
```
extension point com.apple.system_extension.driver_extension failed to start extension with identifier: org.pqrs.Karabiner-DriverKit-VirtualHIDDevice: Error Domain=kernelmanagerd.KMError Code=5 "(null)". Activating anyway.
```

**Comment by omen-works** [comment]
@Seigiard Thanks for the tip to `"Download and install Karabiner from the site"` it fixed my `Error:1` problem with Allowing the Karabiner-VirtualHIDDevice.
